cli/tsc/mod.rs

```diff
@@ -89,28 +68,57 @@ pub static COMPILER_SNAPSHOT: Lazy<Box<[u8]>> = Lazy::new(
 );
 
 pub fn get_types_declaration_file_text(unstable: bool) -> String {
-  let mut types = vec![
-    DENO_NS_LIB,
-    DENO_CONSOLE_LIB,
-    DENO_URL_LIB,
-    DENO_WEB_LIB,
-    DENO_FETCH_LIB,
-    DENO_WEBGPU_LIB,
-    DENO_WEBSOCKET_LIB,
-    DENO_WEBSTORAGE_LIB,
-    DENO_CRYPTO_LIB,
-    DENO_BROADCAST_CHANNEL_LIB,
-    DENO_NET_LIB,
-    SHARED_GLOBALS_LIB,
-    DENO_CACHE_LIB,
-    WINDOW_LIB,
+  let mut assets = get_asset_texts_from_new_runtime()
+    .unwrap()
+    .into_iter()
+    .map(|a| (a.specifier, a.text))
+    .collect::<HashMap<_, _>>();
+
+  let mut lib_names = vec![
+    "deno.ns",
+    "deno.console",
+    "deno.url",
+    "deno.web",
+    "deno.fetch",
+    "deno.webgpu",
+    "deno.websocket",
+    "deno.webstorage",
+    "deno.crypto",
+    "deno.broadcast_channel",
+    "deno.net",
+    "deno.shared_globals",
+    "deno.cache",
+    "deno.window",
   ];
 
   if unstable {
-    types.push(UNSTABLE_NS_LIB);
+    lib_names.push("deno.unstable");
   }
 
-  types.join("\n")
+  lib_names
+    .into_iter()
+    .map(|name| {
+      let asset_url = format!("asset:///lib.{}.d.ts", name);
+      assets.remove(&asset_url).unwrap()
+    })
+    .collect::<Vec<_>>()
+    .join("\n")
+}
+
+fn get_asset_texts_from_new_runtime() -> Result<Vec<AssetText>, AnyError> {
+  // the assets are stored within the typescript isolate, so take them out of there
+  let mut runtime = JsRuntime::new(RuntimeOptions {
+    startup_snapshot: Some(compiler_snapshot()),
+    extensions: vec![Extension::builder("deno_cli_tsc")
+      .ops(get_tsc_ops())
+      .build()],
+    ..Default::default()
+  });
+  let global =
+    runtime.execute_script("get_assets.js", "globalThis.getAssets()")?;
+  let scope = &mut runtime.handle_scope();
```

Comment:
Hopefully `deno types` is not used by a hot path by any user code, this will incur about 20ms startup cost

Reply:
That sounds fine. People could cache it